spec/builtin-history.test.sh

```diff
@@ -32,29 +34,65 @@ history -a
 ^D
 ## END
 
-#### history -r
-rm -f tmp
-echo 'foo' > tmp
 
-echo '
+#### history -w
+
+cd $TMP
+echo 'cmd orig' > tmp
+HISTFILE=tmp
 history -c
+echo 'cmd new' > /dev/null
+history -w
+grep 'orig' tmp > /dev/null
+echo "found=$?"
+## STDOUT:
+found=1
+## END
 
+
+#### history -r 
+# Reads from the history file, and appends it to the current history
+
+cd $TMP
+printf "cmd orig%s\n" {1..10} > tmp
 HISTFILE=tmp
+
+history -c
+
+history -r
 history -r
-history
-' | $SH -i
 
-# match osh's behaviour of echoing ^D for EOF
-case $SH in bash) echo '^D' ;; esac
+history | grep orig | wc -l
 
 ## STDOUT:
-    1  HISTFILE=tmp
-    2  history -r
-    3  foo
-    4  history
-^D
+20
 ## END
 
+
+#### history -n
```

Comment:
likewise "history -n reads new commands from history file, and appends to history in memory"